File: Web.Admin/src/api/services/module/models/EditModuleInput.ts
import { ModuleViewTypeEnum } from "@/api/enums/ModuleViewTypeEnum";
import { CommonStatusEnum } from "@/api/enums/CommonStatusEnum";

/**
 * Fast.Center.Service.Module.Dto.EditModuleInput 编辑模块输入
 */
export interface EditModuleInput {
  /**
   * 模块Id
   */
  moduleId?: number;
  /**
   * 应用Id
   */
  appId?: number;
  /**
   * 模块名称
   */
  moduleName?: string;
  /**
   * 图标
   */
  icon?: string;
  /**
   * 颜色
   */
  color?: string;
  /**
   * 查看类型
   */
  viewType?: ModuleViewTypeEnum;
  /**
   * 排序
   */
  sort?: number;
  /**
   * 状态
   */
  status?: CommonStatusEnum;
  /**
   * 行版本
   */
  rowVersion?: number;
}

File: Api.Server/src/Services/Center/Module/Dto/EditModuleInput.cs
namespace Fast.Center.Service.Menu.Dto;

/// <summary>
/// <see cref="EditModuleInput"/> 编辑模块输入
/// </summary>
public class EditModuleInput : AddModuleInput
{
    /// <summary>
    /// 模块Id
    /// </summary>
    [LongRequired(ErrorMessage = "模块Id不能为空")]
    public long ModuleId { get; set; }

    /// <summary>
    /// 状态
    /// </summary>
    [IntRequired(ErrorMessage = "状态不能为空")]
    public CommonStatusEnum Status { get; set; }

    /// <summary>
    /// 更新版本控制字段
    /// </summary>
    [LongRequired(ErrorMessage = "更新版本控制字段不能为空")]
    public long RowVersion { get; set; }
}